models/issues/review.go

```diff
@@ -618,9 +635,33 @@ func AddReviewRequest(ctx context.Context, issue *Issue, reviewer, doer *user_mo
 		return nil, err
 	}
 
-	// skip it when reviewer hase been request to review
-	if review != nil && review.Type == ReviewTypeRequest {
-		return nil, committer.Commit() // still commit the transaction, or committer.Close() will rollback it, even if it's a reused transaction.
+	if review != nil {
+		// NOTE: in case of failure we still commit the transaction here, else committer.Close() will rollback it,
+		// even if it's a reused transaction.
+
+		// skip it when reviewer hase been request to review
+		if review.Type == ReviewTypeRequest {
+			return nil, committer.Commit()
+		}
+
+		if issue.IsClosed {
+			var err error = ErrReviewRequestOnClosedPR{}
+			if err2 := committer.Commit(); err2 != nil {
+				err = err2
+			}
+			return nil, err
+		}
+
+		if err := issue.LoadPullRequest(ctx); err != nil {
+			return nil, err
+		}
+		if issue.PullRequest.HasMerged {
```

Comment:
Just want to make sure and ask is it safe to assume that the issue is a PR or should I squeeze in a sanity check here somewhere?

Reply:
The check should be on the top of this function whether the `issue.IsPull`.

Reply:
I couldn't find this check above my changes (or within the called functions above them), so I decided to add this check as closed to my code as possible.

Is this alright or should I instead move this check to the top and return an error if it's not true (I guess it acts like an assertion here to make sure we're only calling these functions when dealing with PRs)